tests/infra_test.go

```diff
@@ -665,117 +669,116 @@ var _ = Describe("Infrastructure", func() {
 				Transport: tr,
 			}
 
-			errors := make(chan error)
-			for ix := 0; ix < concurrency; ix++ {
-				go func(ix int) {
-					req, _ := http.NewRequest("GET", metricsURL, nil)
-					resp, err := client.Do(req)
+			for _, url := range metricsURLs {
+				if urlIsIPv6(url) {
+					By("Skip Scraping the Prometheus endpoint (IPv6) until issue https://github.com/kubevirt/kubevirt/issues/4145 is fixed")
+					continue
+				}
+
+				By(fmt.Sprintf("Scraping the Prometheus endpoint %s", url))
+				errors := make(chan error)
+				for ix := 0; ix < concurrency; ix++ {
+					go func(ix int) {
+						req, _ := http.NewRequest("GET", url, nil)
+						resp, err := client.Do(req)
+						if err != nil {
+							fmt.Fprintf(GinkgoWriter, "client: request: %v #%d: %v\n", req, ix, err) // troubleshooting helper
+						} else {
+							resp.Body.Close()
+						}
+						errors <- err
+					}(ix)
+				}
+
+				errorCount := 0
+				for ix := 0; ix < concurrency; ix++ {
+					err := <-errors
 					if err != nil {
-						fmt.Fprintf(GinkgoWriter, "client: request: %v #%d: %v\n", req, ix, err) // troubleshooting helper
-					} else {
-						resp.Body.Close()
+						if urlErr, ok := err.(*netURL.Error); ok {
+							Expect(urlErr.Err.Error()).ToNot(ContainSubstring("connect: network is unreachable"))
```

Comment:
This check is redundant, the next expect already covers it.

Reply:
but i want it to be explicit that the error is due to connectivity
it helps finding errors just this PR found, where ipv6 didnt actually worked,
its a specific interesting error